VK/VK/Presentation/News/ViewControllers/NewsTableViewController.swift

```diff
@@ -64,14 +67,39 @@ final class NewsTableViewController: UITableViewController {
         createCell(indexPath: indexPath)
     }
 
+    override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        let news = news[indexPath.section]
+
+        guard let cellType = NewsCellTypes(rawValue: indexPath.row),
+              cellType == .content,
+              news.attachments.first?.type == .photo,
+              let photo = news.attachments.first?.photo
+        else { return UITableView.automaticDimension }
+        let tableWidth = tableView.bounds.width
+        let cellHeight = (tableWidth * photo.aspectRation) - 200
+        return cellHeight
+    }
+
     // MARK: - Private Method
 
-    private func fetchNews() {
-        networkService.fetchNewsPost { [weak self] result in
+    @objc private func refreshControlAction() {
+        var mostFreshDate: TimeInterval?
+        if let firstItem = news.first {
+            mostFreshDate = firstItem.date + 1
+            fetchNewNews(mostFreshDate: mostFreshDate)
+        }
+    }
+
+    private func fetchNewNews(mostFreshDate: TimeInterval?) {
+        networkService.fetchNewsPost(startTime: mostFreshDate) { [weak self] result in
             guard let self else { return }
             switch result {
             case let .success(response):
-                self.newsFiltered(response)
+                self.refreshControl?.endRefreshing()
+                DispatchQueue.main.async {
```

Comment:
```suggestion
                DispatchQueue.main.async {
```
AF  после выполнения возвращается на главный поток,
после self.refreshControl?.endRefreshing() вызови Thead.сurrent()  проверь, на какой ты потоке